### DISCUSSION TITLE
如何生成插图清单、附表清单？

### DISCUSSION THREAD

**Discussion by ChangPtR** [nju-lug/NJUThesis#245] [Q&A] [answered]
如题，在论文前面生成这两个清单，效果如下。
![1710654778114](https://github.com/nju-lug/NJUThesis/assets/46048008/ef935cda-e0aa-49fd-a4b3-8d40aa8c7eb8)
![1710654847156](https://github.com/nju-lug/NJUThesis/assets/46048008/fc22e75d-0152-40a2-9096-61744ae34406)
![1710654889010](https://github.com/nju-lug/NJUThesis/assets/46048008/2f5118ef-3260-4de9-9bb4-417c655d06bc)

**Comment by atxy-blip** [comment, marked as answer]
`\listoffigures` `listoftables`，见 `njuthesis`手册 3.7.4 节。

**Comment by acddca** [comment]
如何修改自动生成的目录里的内容，如在正文中图1名为“跳水示意图” 在插图目录中想将其命名为“示意图”

**Reply by atxy-blip** [reply]
这是 `\caption` 的附加用法：
```tex
\caption[示意图]{跳水示意图}
```

**Comment by acddca** [comment]
请问目录中怎么去掉“目录和相应页码” 但保留“插图清单和附表清单以及相应页码”

**Reply by atxy-blip** [reply]
见 `njuthesis-setup.def`
```tex
\njusetup{
    tableofcontents/toc-entry = false,
    listoffigures/toc-entry   = true,
    listoftables/toc-entry    = true
}
```
但实际上逻辑写出了一点问题，稍后改一下发个新版。

**Reply by acddca** [reply]
请问实际上对于毕业论文中“目录中”可以包含“目录”这个条目吗？不知道大多数老师是怎么要求的？

**Reply by acddca** [reply]
我改了下设置还是不行，会去掉所有的三个...如上所问，如果其实可以包含“目录”这个条目 我就不改了：）
\njusetup{
    tableofcontents/toc-entry = false,
    listoffigures/toc-entry   = true,
    listoftables/toc-entry    = true
}

**Reply by atxy-blip** [reply]
> 我改了下设置还是不行，会去掉所有的三个

可以先试用新版本，把联动的设置项删了：<https://github.com/nju-lug/NJUThesis/releases/tag/v1.4.2> 还没发布是因为 CI 莫名其妙挂了没找到原因 orz

**Comment by acddca** [comment]
<img width="542" alt="截图" src="https://github.com/user-attachments/assets/7e95f1be-1a8e-4c78-90b7-bdd1c81be319">
